src/components/welcome/MyAccounts/SubAccountItem.tsx

```diff
@@ -0,0 +1,138 @@
+import { LoopIcon } from '@/features/counterfactual/CounterfactualStatusButton'
+import { selectUndeployedSafe } from '@/features/counterfactual/store/undeployedSafesSlice'
+import type { ChainInfo, SafeOverview } from '@safe-global/safe-gateway-typescript-sdk'
+import { useCallback, useMemo } from 'react'
+import { ListItemButton, Box, Typography, Chip, Skeleton } from '@mui/material'
+import Link from 'next/link'
+import SafeIcon from '@/components/common/SafeIcon'
+import Track from '@/components/common/Track'
+import { OVERVIEW_EVENTS, OVERVIEW_LABELS } from '@/services/analytics'
+import { AppRoutes } from '@/config/routes'
+import { useAppSelector } from '@/store'
+import { selectChainById } from '@/store/chainsSlice'
+import css from './styles.module.css'
+import { selectAllAddressBooks } from '@/store/addressBookSlice'
+import SafeListContextMenu from '@/components/sidebar/SafeListContextMenu'
+import useSafeAddress from '@/hooks/useSafeAddress'
+import useChainId from '@/hooks/useChainId'
+import { sameAddress } from '@/utils/addresses'
+import classnames from 'classnames'
+import { useRouter } from 'next/router'
+import ErrorOutlineIcon from '@mui/icons-material/ErrorOutline'
+import type { SafeItem } from './useAllSafes'
+import FiatValue from '@/components/common/FiatValue'
+import QueueActions from './QueueActions'
+
+type SubAccountItem = {
+  safeItem: SafeItem
+  safeOverview?: SafeOverview
+  onLinkClick?: () => void
+}
+
+const SubAccountItem = ({ onLinkClick, safeItem, safeOverview }: SubAccountItem) => {
+  const { chainId, address } = safeItem
+  const chain = useAppSelector((state) => selectChainById(state, chainId))
+  const undeployedSafe = useAppSelector((state) => selectUndeployedSafe(state, chainId, address))
+  const safeAddress = useSafeAddress()
+  const currChainId = useChainId()
+  const router = useRouter()
+  const isCurrentSafe = chainId === currChainId && sameAddress(safeAddress, address)
+  const isWelcomePage = router.pathname === AppRoutes.welcome.accounts
+  const isSingleTxPage = router.pathname === AppRoutes.transactions.tx
+
+  const trackingLabel = isWelcomePage ? OVERVIEW_LABELS.login_page : OVERVIEW_LABELS.sidebar
+
+  /**
+   * Navigate to the dashboard when selecting a safe on the welcome page,
+   * navigate to the history when selecting a safe on a single tx page,
+   * otherwise keep the current route
+   */
+  const getHref = useCallback(
```

Comment:
We could make this a hook since it's used again in the other account item component